presto-native-execution/presto_cpp/main/PrestoServerOperations.cpp

```diff
@@ -185,22 +189,42 @@ std::string PrestoServerOperations::veloxQueryConfigOperation(
   return unsupportedAction(op);
 }
 
-std::string PrestoServerOperations::debugOperation(
+std::string PrestoServerOperations::taskOperation(
     const ServerOperation& op,
     proxygen::HTTPMessage* message) {
+  if (taskManager_ == nullptr) {
+    return "Task Manager not found";
+  }
+  const auto taskMap = taskManager_->tasks();
   switch (op.action) {
-    case ServerOperation::Action::kTask: {
+    case ServerOperation::Action::kGetDetail: {
       const auto id = message->getQueryParam("id");
-      if (!taskManager_) {
-        return "Task Manager not found";
-      }
-      const auto& map = taskManager_->tasks();
-      const auto& task = map.find(id);
-      if (task == map.end()) {
+      const auto& task = taskMap.find(id);
+      if (task == taskMap.end()) {
         return fmt::format("No task found with id {}", id);
       }
       return task->second->toJsonString();
     }
+    case ServerOperation::Action::kListAll: {
+      std::stringstream oss;
+      oss << "[";
+      for (const auto& task : taskMap) {
+        oss << task.first << "(" << taskStateString(task.second->task->state())
+            << "),\n";
```

Comment:
Imagine you run this for a cluster of 1000 workers.... Having 1 line per task would be overkill.

Reply:
We could pass count with zero to fetch all, otherwise stop at the specified count.

Reply:
This is not supposed to be run on all workers at the same time. But yeah we can have that count feature